src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs

```diff
@@ -128,9 +128,13 @@ internal List<SourceRepository> GetEffectiveSources(ISettings settings, IList<Pa
 
         private List<SourceRepository> GetEffectiveSourcesCore(ISettings settings, IList<PackageSource> dgSpecSources)
         {
-            var sourceObjects = dgSpecSources.ToDictionary(k => k.Source, v => v, StringComparer.Ordinal);
             var packageSourceProvider = new PackageSourceProvider(settings);
             var packageSourcesFromProvider = packageSourceProvider.LoadPackageSources();
+            var sourceObjects = new Dictionary<string, PackageSource>();
+            foreach(var source in dgSpecSources)
```

Comment:
nit: since `dgSpecSources` is a list then you can further improve it by replacing `ForEach` with `For`. Which will avoid `Enumeration` altogether from this. It might not make much difference, but just a good perf practice

Reply:
That's a good point. 
It doesn't help us much, but it's a good practice.